SimonsAlgorithm/Tests.cs

```diff
@@ -46,20 +46,16 @@ public Instance(List<List<long>> transformation, List<long> kernel, int instance
 
             public BooleanVector Kernel => new BooleanVector(kernel);
 
-            public QArray<QArray<long>> Transformation => new QArray<QArray<long>>(
+            public IQArray<IQArray<long>> Transformation => new QArray<IQArray<long>>(
                 transformation.Select(
                     vector => new QArray<long>(vector)));
 
-            public QArray<QArray<long>> ExtendedTransformation
+            public IQArray<IQArray<long>> ExtendedTransformation
             {
                 get
                 {
-                    var array = new QArray<QArray<long>>(
-                        transformation.Select(
-                            vector => new QArray<long>(vector)))
-                    {
-                        new QArray<long>(transformation.Last())
-                    };
+                    var array = (IQArray<IQArray<long>>)new QArray<IQArray<long>>(transformation.Select(vector => new QArray<long>(vector)));
```

Comment:
Could you please keep this split into multiple lines for readabilty?

Reply:
sure